tests/components/energyzero/test_services.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for the sensors provided by the EnergyZero integration."""
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.energyzero.const import DOMAIN, SERVICE_NAME
+from homeassistant.core import HomeAssistant
+
+pytestmark = [pytest.mark.freeze_time("2022-12-07 15:00:00")]
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_has_service(
+    hass: HomeAssistant,
+) -> None:
+    """Test the existence of the EnergyZero Service."""
+    assert hass.services.has_service(DOMAIN, SERVICE_NAME)
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.parametrize("price_type", ["gas", "energy"])
+@pytest.mark.parametrize("incl_btw", [True, False, None])
+@pytest.mark.parametrize("start", ["2023-01-01 00:00:00", "incorrect date", None])
+@pytest.mark.parametrize("end", ["2023-01-01 00:00:00", "incorrect date", None])
+async def test_service(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    price_type: str,
+    incl_btw: bool,
+    start: str,
+    end: str,
+) -> None:
+    """Test the EnergyZero Service."""
+
+    data = {
+        "type": price_type,
+    }
+
+    if incl_btw is not None:
+        data["incl_btw"] = incl_btw
+    if start is not None:
+        data["start"] = start
+    if end is not None:
+        data["end"] = end
```

Comment:
In here you can do `data |= incl_btw`

Reply:
Reason behind this is that we want to avoid branches in tests so we don't test a "maybe"